cmake/go.cmake

```diff
@@ -48,14 +48,25 @@ function(go_executable dest package)
     endif()
 
     set(depfile "${CMAKE_CURRENT_BINARY_DIR}/${dest}.d")
-    add_custom_command(OUTPUT ${dest}
-                       COMMAND ${GO_EXECUTABLE} build
-                               -o ${CMAKE_CURRENT_BINARY_DIR}/${dest} ${package}
-                       COMMAND ${GO_EXECUTABLE} run ${godeps} -format depfile
-                               -target ${target} -pkg ${package} -out ${depfile}
-                       WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
-                       DEPENDS ${godeps} ${CMAKE_SOURCE_DIR}/go.mod
-                       DEPFILE ${depfile})
+
+    if(CMAKE_VERSION VERSION_LESS "3.7")
+      add_custom_command(OUTPUT ${dest}
+                         COMMAND ${GO_EXECUTABLE} build
+                         -o ${CMAKE_CURRENT_BINARY_DIR}/${dest} ${package}
+                         COMMAND ${GO_EXECUTABLE} run ${godeps} -format depfile
+                         -target ${target} -pkg ${package} -out ${depfile}
+                         WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+                         DEPENDS ${godeps} ${CMAKE_SOURCE_DIR}/go.mod)
```

Comment:
Might be a dumb question, but does this version work if `CMAKE_VERSION` is greater than `3.7`? And did we need `DEPFILE ${depfile}` on L68 below?

Reply:
Oh good catch I accidentally left the keyword off 🤦 . The second `COMMAND` directive is just generating a file for ninja-build to consume so it knows what component files make up the first command, so that if those files change it can rerun the commands. Really it would only impact a developer of AWS-LC who is using a very old version of CMake....so highly unlikely.